02_housing/0_housing_microdata.qmd

```diff
@@ -107,54 +116,105 @@ vacant_microdata22 <- define_extract_usa(
   ) %>% arrange(statefip, puma)
 
 # save vacancy data
-write_csv(vacant_microdata22,  here::here("02_housing","data","temp","vacancy_microdata2022.csv"))
+write_csv(vacant_microdata,  here::here("02_housing","data","temp",paste0("vacancy_microdata",year,".csv")))
+}
+
+# extract data for 2022 and 2021
+map(county_years, extract_vacant_ipums)
 ```
 
 ## 2) Prepare crosswalks
 
 #### 2a) Place crosswalk
 
-```{r}
+**NOTE:** In future years of updates, be sure to check the puma/place crosswalk. Some PUMAs change over the years so make sure that when you merge the PUMA microdata and place crosswalk you are getting the expected number of places (in 2024 update it was 486 places)
 
+```{r}
 # Prepare the Census Place to PUMA crosswalk
-puma_place_2022 <- read_csv(here::here("geographic-crosswalks/data/crosswalk_puma_to_place.csv")) %>% 
-  filter(crosswalk_period == 2022)
+# open relevant crosswalk data
+puma_place_crosswalk <- function(year){
+  # read in crosswalk file and filter to relevant year
+  puma_place <- read_csv(here::here("geographic-crosswalks/data/crosswalk_puma_to_place.csv")) %>% 
+  # NOTE: crosswalk has "2022", and "pre-2022" for year, in future updates, make sure this column is named similarly
+    mutate(crosswalk_period = if_else(crosswalk_period == "pre-2022", "2021", crosswalk_period)) %>% 
+    filter(crosswalk_period == year)
+  
+  # save a version with just the place-level values of data quality variables 
+  # NOTE: data quality variables calculated in geographic-crosswalks/generate_puma_place_crosswalks.qmd
+  place_puma <- puma_place %>%
+    group_by(statefip, place) %>%
+    summarize(puma_flag = mean(geographic_allocation_quality))
+  
+  # save puma_place file
+  write_csv(puma_place,  here::here("02_housing","data","temp",paste0("puma_place_", year,".csv")))
+  
+  # save place_puma file
+  write_csv(place_puma,  here::here("02_housing","data","temp",paste0("place_puma_", year,".csv")))
+}
 
-# check number of unique places in data
-puma_place_2022 %>% distinct(statefip, place) %>% nrow() # 486 observations
+# loop over 2021 and 2022
+map(place_years, puma_place_crosswalk)
 
-# save a version with just the place-level values of data quality variables 
-# NOTE: data quality variables calculated in geographic-crosswalks/generate_puma_place_crosswalks.qmd
-place_puma <- puma_place_2022 %>%
-  group_by(statefip, place) %>%
-  summarize(puma_flag = mean(geographic_allocation_quality))
+# NOTE TO JP: 2021 is missing data for place fips 72122 in Georgia
```

Comment:
Thanks for pointing this out, turns out that city in Georgia was incorporated in 2016 and the pre-2022 crosswalk uses places from 2014 so it's not being captured. Given the limitaitons from GeoCorr I don't think we can manually add it back in.